app/src/main/java/com/sun/weatherapp/screen/music/MusicPresenter.kt

```diff
@@ -0,0 +1,108 @@
+package com.sun.weatherapp.screen.music
+
+import com.sun.weatherapp.data.model.Artist
+import com.sun.weatherapp.data.model.MusicTabType
+import com.sun.weatherapp.data.model.Song
+import com.sun.weatherapp.screen.base.BasePresenter
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class MusicPresenter : BasePresenter<MusicContract.View>(), MusicContract.Presenter {
+    
+    private var currentTab = MusicTabType.RECOMMEND
+    private var isInitialLoad = true
+    
+    override fun loadMusicData(tabType: MusicTabType) {
+        currentTab = tabType
+        
+        presenterScope.launch {
+            try {
+                if (isInitialLoad) {
+                    getView()?.showLoading()
+                }
+                
+                getView()?.updateSelectedTab(tabType)
+                delay(500)
+                when (tabType) {
+                    MusicTabType.RECOMMEND -> {
+                        getView()?.showSongs(getRecommendSongs())
+                    }
+                    MusicTabType.ARTIST -> {
+                        getView()?.showArtists(getMockArtists())
+                    }
+                    MusicTabType.ALL_SONGS -> {
+                        getView()?.showSongs(getAllSongs())
+                    }
+                }
+                
+                getView()?.hideLoading()
+                isInitialLoad = false
+                
+            } catch (e: Exception) {
+                getView()?.hideLoading()
+                getView()?.showError("Không thể tải dữ liệu nhạc: ${e.message}")
+                isInitialLoad = false
+            }
+        }
+    }
+    
+    override fun onTabSelected(tabType: MusicTabType) {
+        if (currentTab != tabType) {
+            loadMusicData(tabType)
+        }
+    }
+    
+    override fun onArtistClicked(artist: Artist) {
+        // TODO: Navigate to artist detail
+    }
+    
+    override fun onSongClicked(song: Song) {
+        // TODO: Play song
```

Comment:
Thêm logic xử lý hàm này cho view đi
```suggestion
        getView()?.navigateTo...
```